sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/VolumeEvaluationStage.java

```diff
@@ -25,24 +25,61 @@
     private final String taskName;
     private final Optional<String> resourceId;
     private final boolean useDefaultExecutor;
+    private final Optional<String> sourceRoot;
 
-    public VolumeEvaluationStage(
+    public static VolumeEvaluationStage getNew(VolumeSpec volumeSpec, String taskName, boolean useDefaultExecutor) {
+        return new VolumeEvaluationStage(
+                volumeSpec,
+                taskName,
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                useDefaultExecutor);
+    }
+
+    public static VolumeEvaluationStage getExisting(
+            VolumeSpec volumeSpec,
+            String taskName,
+            Optional<String> resourceId,
+            Optional<String> persistenceId,
+            Optional<String> sourceRoot,
+            boolean useDefaultExecutor) {
+        return new VolumeEvaluationStage(
+                volumeSpec,
+                taskName,
+                resourceId,
+                persistenceId,
+                sourceRoot,
+                useDefaultExecutor);
+    }
+
+    private VolumeEvaluationStage(
             VolumeSpec volumeSpec,
             String taskName,
             Optional<String> resourceId,
             Optional<String> persistenceId,
+            Optional<String> sourceRoot,
             boolean useDefaultExecutor) {
         this.volumeSpec = volumeSpec;
         this.taskName = taskName;
         this.resourceId = resourceId;
         this.persistenceId = persistenceId;
+        this.sourceRoot = sourceRoot;
         this.useDefaultExecutor = useDefaultExecutor;
     }
 
     private boolean createsVolume() {
         return !persistenceId.isPresent();
     }
 
+    private static boolean isMountVolume(MesosResource mesosResource) {
```

Comment:
Duplicate of `ResourceUtils.isMountVolume()`. Might not even be used? Diff doesn't show anyone calling into this.